metrics/web/docs/features/builds_aggregation/diagrams/build_days_data_layer_class_diagram.puml

```diff
@@ -0,0 +1,43 @@
+@startuml build_days_data_layer_class_diagram
+
+package common {
+    package domain {
+        package repositories {
+            interface BuildDayRepository {
+                + projectBuildDaysFromDateStream(String projectId, DateTime from) : Stream<List<BuildDay>>
+            }
+        }
+
+        package entities {
+            class BuildDay {
+                + projectId : String
+                + successful: int
+                + failed: int
+                + unknown: int
+                + inProgress: int
+                + totalDuration : int
+                + day : Timestamp
+            }
+        }
+    }
+
+    package data {
+        package repositories as data_repositories {
+            class FirestoreBuildDayRepository {}
+        }
+
+        package deserializer {
+            class BuildDayDataDeserializer {
```

Comment:
I believe it would be better to create a `BuildDayData` class that will implement the `DataModel` class.